In some situations Emacs' default Pure Storage is not big enough. This leads to warnings like
Warning Warning!!!  Pure space overflow    !!!Warning Warning

The Lisp Manual at https://www.gnu.org/software/emacs/manual/html_node/elisp/Pure-Storage.html
recommends to increase the pure storage size during compilation time which is what we do with this patch.
--- emacs-30.1/src/puresize.h.orig
+++ emacs-30.1/src/puresize.h
@@ -39,7 +39,7 @@
    amount of storage.  This is a lot more update-robust that defining
    BASE_PURESIZE or even PURESIZE directly.  */
 #ifndef SYSTEM_PURESIZE_EXTRA
-#define SYSTEM_PURESIZE_EXTRA 0
+#define SYSTEM_PURESIZE_EXTRA 1000000
 #endif
 
 #ifndef SITELOAD_PURESIZE_EXTRA
